src/mgmt/srv_pool.c

```diff
@@ -588,32 +587,49 @@ ds_mgmt_destroy_pool(uuid_t pool_uuid, const char *group, uint32_t force)
 	if (rc != 0) {
 		if (rc == -DER_ALREADY)
 			rc = 0;
+		goto out_ranks;
+	}
+
+	rc = pool_get_svc_ranks(svc, pool_uuid, &psvcranks);
+	if (rc != 0) {
+		D_ERROR("Failed to get pool service ranks "DF_UUID" rc: %d\n",
+			DP_UUID(pool_uuid), rc);
 		goto out_svc;
 	}
 
+	/* Check active pool connections, evict only if force */
+	rc = ds_pool_svc_check_evict(pool_uuid, psvcranks, force);
```

Comment:
Yes I think I need to restore the order so that this comes first. Otherwise the pool's pr_state could be left at PR_DESTROYING even when connections exist and we should just avoid any of the remaining steps to destroy the pool.